SU2_CFD/src/output/COutput.cpp

```diff
@@ -248,6 +248,25 @@ void COutput::SetHistory_Output(CGeometry *geometry,
   Convergence_Monitoring(config, curInnerIter);
 
   Postprocess_HistoryData(config);
+
+  /*--- Output for multizone problems (using only the master node) ---*/
+
+  bool write_header, write_history, write_screen;
+  if (rank == MASTER_NODE && !noWriting) {
+
+    /*--- Write the history file ---------------------------------------------------------------------------*/
+    write_history = WriteHistoryFile_Output(config);
+    if (write_history) SetHistoryFile_Output(config);
+
+    /*--- Write the screen header---------------------------------------------------------------------------*/
+    write_header = WriteScreen_Header(config);
+    if (write_header) SetScreen_Header(config);
+
+    /*--- Write the screen output---------------------------------------------------------------------------*/
+    write_screen = WriteScreen_Output(config);
+    if (write_screen) SetScreen_Output(config);
+
+  }
```

Comment:
This block of code is showing up quite a bit, make it a private function (like Postprocess_HistoryData) please.

Reply:
Yes agreed, I've added a function, OutputScreenAndHistory, which takes care of these.